lib/packages.js

```diff
@@ -0,0 +1,223 @@
+'use strict';
+
+// requires
+var Log = require('log');
+var testing = require('testing');
+var db = require('./db.js');
+var config = require('../config.js');
+
+// globals
+var packagesCollection;
+var log = new Log(config.logLevel);
+
+
+function openCollection(callback)
+{
+	if (packagesCollection)
+	{
+		return callback(null, packagesCollection);
+	}
+	db.addCallback(function(error, result)
+	{
+		if (error)
+		{
+			return callback('Could not access database: ' + error);
+		}
+		packagesCollection = result.collection(config.packagesCollection);
+		if (packagesCollection)
+		{
+			console.log('was there');
+			return callback(null, packagesCollection);
+		}
+			console.log('creating');
+		result.createCollection(config.packagesCollection, function(error)
+		{
+			if (error)
+			{
+				return callback(error);
+			}
+			return callback(null, result.collection(config.packagesCollection));
+		});
+	});
+}
+
+function testOpenCollection(callback)
+{
+	packagesCollection = null;
+	db.setTestMode(function(error)
+	{
+		testing.check(error, 'Could not set test mode', callback);
+		// init
+		openCollection(function(error, collection)
+		{
+			testing.check(error, 'Could not open collection', callback);
+			testing.assert(collection, 'Did not open collection', callback);
+			testing.success(callback);
+		});
+	});
+}
+
+exports.update = function(object, callback)
+{
+	openCollection(function(error, collection)
+	{
+		if (error)
+		{
+			log.error('Could not open collection: %s', error);
+			return callback('database not available');
+		}
+		console.log('collection: %s', collection);
```

Comment:
I was just debugging an issue with tests, done in commit 9600516cc052617a69b5d550639623149246a49f.